test/config-luatest/vshard_test.lua

```diff
@@ -652,3 +652,89 @@ g.test_no_suitable_uri = function(g)
     local err = 'No suitable URI provided for instance "instance-001"'
     t.assert_str_contains(res.stderr, err)
 end
+
+-- Make sure that rebalancing will be disabled if rebalancer_mode == 'off', even
+-- if rebalancer sharding role is assigned.
+g.test_rebalancer_mode = function(g)
+    t.skip_if(not has_vshard, 'Module "vshard" is not available')
+    local dir = treegen.prepare_directory(g, {}, {})
+    local config = [[
+    credentials:
+      users:
+        guest:
+          roles: [super]
+        storage:
+          roles: [sharding]
+          password: "storage"
+
+    iproto:
+      listen:
+        - uri: 'unix/:./{{ instance_name }}.iproto'
+      advertise:
+        sharding:
+          login: 'storage'
+
+    sharding:
+      rebalancer_mode: off
+
+    groups:
+      group-001:
+        replicasets:
+          replicaset-001:
+            sharding:
+              roles: [storage, rebalancer, router]
+            instances:
+              instance-001: {}
+    ]]
+    local config_file = treegen.write_script(dir, 'config.yaml', config)
+    local opts = {
+        env = {LUA_PATH = os.environ()['LUA_PATH']},
+        config_file = config_file,
+        chdir = dir,
+        alias = 'instance-001',
+    }
+    g.server = server:new(opts)
+    g.server:start()
+
+    -- Check vshard config on each instance.
+    local exp = {
+        box_cfg_mode = "manual",
+        bucket_count = 3000,
+        discovery_mode = "on",
+        failover_ping_timeout = 5,
+        identification_mode = 'name_as_key',
+        rebalancer_disbalance_threshold = 1,
+        rebalancer_max_receiving = 100,
+        rebalancer_max_sending = 1,
+        rebalancer_mode = "off",
+        sched_move_quota = 1,
+        sched_ref_quota = 300,
+        schema_management_mode = "manual_access",
+        shard_index = "bucket_id",
+        sync_timeout = 1,
+        sharding = {
+            ["replicaset-001"] = {
+                master = "auto",
+                rebalancer = true,
+                replicas = {
+                    ["instance-001"] = {
+                        uri = {
+                            login = "storage",
+                            password = "storage",
+                            uri = "unix/:./instance-001.iproto",
+                        },
+                    },
+                },
+                weight = 1,
+            },
+        },
+    }
```

Comment:
I would compare only values that we want to check. Reasons:

1. It helps to understand the test case.
2. The test case is immutable for additions to vshard's configuration (so it will likely work without problem on a newer vshard version).

Reply:
Thanks, fixed.